src/AvroConvert/AvroObjectServices/BuildSchema/AvroContractResolver.cs

```diff
@@ -182,20 +182,21 @@ internal MemberSerializationInfo[] ResolveMembers(Type type)
             }
 
             var members = membersToSerialize
-            .Select(m =>
-            {
-                var customAttributes = m.GetCustomAttributes(false);
-
-                return new
+                .Where(x => !x.GetCustomAttributes(false).OfType<IgnoreDataMemberAttribute>().Any())
```

Comment:
Please extract the result of GetCustomAttributes() to the variable and use it wherever any custom attribute is needed

Reply:
Thanks, please see updated changes. Happy to refactor further if needed.